### DISCUSSION TITLE
"open supported links" setting in Android app reverts to default value on every update

### DISCUSSION THREAD

**Discussion by Pika-Pool** [community/community#18126] [Mobile]
"open supported links" setting in Android app reverts to default value of "allow app to open supported links", every time the app is updated. I have to repeatedly set its value to "ask everytime".

**Comment by brianlovin** [comment]
Thanks for the note @Pika-Pool — it sounds like this might be something happening at the Android OS level, not at the GitHub level. Are you running into this issue with any other app?

**Reply by Pika-Pool** [reply]
No, the issue is only on the GitHub app. I'm not facing the issue in any other apps I've installed

**Reply by brianlovin** [reply]
Got it. We have been trying to see if there's a way to reproduce this incorrect behavior, but haven't managed to find any issues so far. We have an open issue to see if we can figure out what might be going on, and to see if other people report experiencing the same bug in the future. Thanks @Pika-Pool

**Reply by alcere** [reply]
Hi @Pika-Pool, the issue you described looks similar to an existing bug in the Android OS: https://issuetracker.google.com/issues/204233247. 
Could you let us know if the issue persists? If so, are there any other apps affected by the same besides GitHub?
Thanks!

**Reply by Pika-Pool** [reply]
Hi @alcere, the issue persisted the last time I updated the app a couple of days ago. I haven't noticed this issue affecting any other apps besides GitHub though.

**Reply by alcere** [reply]
That's good to know, thanks! Can you share with us the device and Android OS version you are using please?

**Reply by Pika-Pool** [reply]
Device: OnePlus 7
Android OS version: 11
Oxygen OS 11.0.9.1

**Reply by alcere** [reply]
Hi @Pika-Pool, sorry for (very) late response! The team added recently some changes to the code managing the permissions, did this problem got fixed by any chance?

**Reply by Pika-Pool** [reply]
Hi @alcere, unfortunately I've stopped using the GitHub app as of late. I've downloaded it again to check this. One i get an update through Google Play, I'll update this thread with the result.

If there is any other way to test this issue, without waiting for an update on Google Play, please let me know.

**Reply by alcere** [reply]
No problem, thanks for verifying! The easiest way is through an update, one should arrive later this week.

**Reply by Pika-Pool** [reply]
@alcere, the problem still persists when I updated the app this time. Latest version of GitHub Mobile: V1.95.0(700)

**Reply by alcere** [reply]
Thanks for verifying! We will keep working on this then.

**Reply by Pika-Pool** [reply]
I have these versions now:
Android v12
OxygenOS v12.1
Github v1.106.1

And the original issue seems to be fixed. I updated a couple of days back, and the setting "open supported links" under "open by default" has the same state(it was set off before update and it remained off).

But now there's no options for ask everytime, so it doesn't ask me to open in app or I'm browser. Just opens the likes in browser by default

**Comment by rayc680726** [comment]
3JTe9sfPLgPVzuqEdn1Ykichtv67ZupimU

**Comment by Azucena811** [comment]
![Screenshot_20230114-194938_PDF Reader.jpg](https://user-images.githubusercontent.com/120296644/212516378-00d1932f-0211-4048-b10e-a68f647dbad0.jpg)

**Comment by ghost** [comment]
> "open supported links" setting in Android app reverts to default value on "allow app to open supported links", every time the app is updated. I have to repeatedly set its value to "ask everytime".

**Comment by ghost** [comment]
> Thanks for the note @Pika-Pool — it sounds like this might be something happening at the Android OS level, not at the GitHub level. Are you running into this issue with any other app?
